### DISCUSSION TITLE
Pico SDK / Adafruit USB stack use

### DISCUSSION THREAD

**Discussion by artouste** [earlephilhower/arduino-pico#358] [Q&A]
Hello
I saw that the pico2040 port has been upgraded to version 1.97
After a lot of testing I can't quite understand the logic of use

USB stack between adafruit stack and PICO Stack
My problem is simple but I can't do it, probably because of a bad understanding on the part.

To put it simply: I am trying to recover MIDI input and generate Mouse and / or Keyboard events as output.

But I can't seem to link MIDI and MOUSE / Keyboard together

thank you for clarifying me on my incomprehensions.

hard copy of my last tests 

https://www.cjoint.com/doc/21_12/KLcmUyjjFQE_hcopy024.jpg

**Comment by earlephilhower** [comment]
I would suggest just going with the Adafruit stack and looking at some of their included examples.  The `Mouse` and `Keyboard` libraries require the Pico SDK and don't have any special-casing for non-HID types.  Good luck!
